learning/tests.py

```diff
@@ -27,6 +34,71 @@ def setUp(self):
         logged_in = self.client.login(**self.credentials)
 
     def test_access_to_check_user_skills_with_login(self):
-        response = self.client.get("/learning/check_user_skills/1", follow=True)
+        response = self.client.post("/learning/check_user_skills/1", follow=True)
         self.assertEquals(response.status_code, 200)
         self.assertTrue(response.context["user"].is_authenticated)
+
+    def test_show_user_when_authenticated(self):
+        response = self.client.get("/learning/", follow=True)
+        self.assertEquals(response.status_code, 200)
+        self.assertTrue(response.context["user"])
+
+
+class LearningTest(TestCase):
+    def setUp(self):
+        self.credentials = {"username": "testuser", "password": "secret"}
+        self.user = User.objects.create_user(
+            username=self.credentials.get("username"), password=self.credentials.get("password")
+        )
+        self.flashcard_id = 1
+        logged_in = self.client.login(**self.credentials)
+
+    def test_create_new_answer_if_not_exists(self):
+        data = {"is_correct": "true", "flashcard_id": self.flashcard_id}
+        flashcard = Flashcard.objects.get(id=self.flashcard_id)
+        response = self.client.post("/learning/save_answer/", data=data, follow=True)
+        self.assertEquals(response.status_code, 200)
+        self.assertTrue(Answer.objects.filter(user_id=self.user.id, flashcard=flashcard).exists())
+
+    def save_user_answer(self, answer):
```

Comment:
pomysł dobry, żeby zrobić z tego jedną funkcję, ale przez te rozbudowane ify to brzdko wygląda, więc bym przeniósł to do test_save_correct_answer i test_save_incorrect_answer